Lib/Src/CJsonParser.cpp

```diff
@@ -0,0 +1,339 @@
+/**
+ * @file CJsonParser.cpp
+ * @brief This is a JSON parser that includes a tokeniser (view getTokens()) and
+ * a parser. The tokeniser breaks up a string into JSON components. The parser
+ * checks if the components follow JSON syntax and returns true if it does.
+ */
+/*
+ * Created on : Apr 4, 2022
+ * Author : niuslar
+ */
+
+#include "CJsonParser.h"
+
+#define DEFAULT_VALUE (0.00)
+
+/**
+ * @brief Constructor
+ */
+CJsonParser::CJsonParser() {}
+
+/**
+ * @brief Parse string
+ * @param String to be parsed
+ * @return True if a JSON object was recognised
+ */
+bool CJsonParser::parse(const etl::string<MAX_STRING_SIZE> &string)
+{
+    // Reset command variables
+    m_argument_counter = 0;
+    m_command_name = "";
+
+    getTokens(string);
+    m_token_index = 0;
+    return parseObject();
+}
+
+/**
+ * @brief Tokeniser* that scans string and breaks it up into tokens
+ *
+ * @param String
+ *
+ * @note A tokeniser or lexer takes a string and breaks it up into
+ * tokens. A token is the basic building block of the language. For example, in
+ * English, single words and punctuation marks would be tokens.
+ */
+void CJsonParser::getTokens(const etl::string<MAX_STRING_SIZE> &string)
+{
+    // Reset variables
+    m_tokens.clear();
+    m_token_counter = 0;
+    token_t current_token;
+
+    // Loop through characters in string with a range-base for-loop
+    // Note: To learn about range-based loops:
+    // https://www.techiedelight.com/iterate-over-characters-string-cpp/
+    for (char current_char : string)
+    {
+        switch (current_char)
+        {
+            case '0':
+            case '1':
+            case '2':
+            case '3':
+            case '4':
+            case '5':
+            case '6':
+            case '7':
+            case '8':
+            case '9':
+                if (current_token.type == STRING ||
```

Comment:
These extended processing inside `case:` statements should be inside separate methods with good descriptive names. Example: `processDigit()`, `processDecimalPoint()`, `processSpacer()`.